### PR TITLE
api: validate selection on create

### PR DESCRIPTION
**Pull request by hachreak**
Signed-off-by: Leonardo Rossi <leonardo.r@cern.ch>

### CODE REVIEW

File: invenio_deposit/api.py [hunk 1]
```diff
@@ -341,7 +341,7 @@ def publish(self, pid=None, id_=None):
             self._publish_new(id_=id_)
         else:  # Update after edit
             record = self._publish_edited()
-            record.commit()
+            record.commit(**kwargs)
```
Comment (jirikuncar) [inline invenio_deposit/api.py]: We should think if we need to have two separate groups of keyword arguments for record and deposit (`self`).

Comment (hachreak) [inline invenio_deposit/api.py]: what do you think if it's the same?

Comment (jirikuncar) [inline invenio_deposit/api.py]: For consistency reasons, I would say that all `kwargs` should go to method binded to `self`. In this specific case we can add new argument `record_kwargs`, `record_ctx`, or similar. Then the code should be something like `record.commit(**record_kwags or {})`.

Comment (hachreak) [inline invenio_deposit/api.py]: I'll remove this change on `publish()` (it's not really needed). 

p.s. I'm trying to write some tests but I need `invenio-records` is released with latest changes.
Could you please release it while I'm finish to create the tests? Thanks! :+1:

**Comment by egabancho** [comment]
https://github.com/inveniosoftware/invenio-deposit/pull/166

**Pull request closed without merging**